src/com/hms_networks/americas/sc/localdatafiles/LocalDataFileManager.java

```diff
@@ -0,0 +1,476 @@
+      String alarmStatus,
+      String utcTimestamp,
+      String localTimestamp) {
+    // Get tag type
+    TagType tagType = TagInfoManager.getTagInfoArray()[tagID].getType();
+
+    // Append UTC time and delimiter
+    localDataFileWriteBuffer.append(utcTimestamp);
+    localDataFileWriteBuffer.append(LocalDataFileConstants.LOCAL_DATA_FILE_DELIMITER);
+
+    // Append local time and delimiter
+    localDataFileWriteBuffer.append(localTimestamp);
+    localDataFileWriteBuffer.append(LocalDataFileConstants.LOCAL_DATA_FILE_DELIMITER);
+
+    // Append row type and delimiter
+    localDataFileWriteBuffer.append(LocalDataFileConstants.LOCAL_DATA_FILE_ROW_TYPE_ALARM);
+    localDataFileWriteBuffer.append(LocalDataFileConstants.LOCAL_DATA_FILE_DELIMITER);
+
+    // Append tag name and delimiter
+    localDataFileWriteBuffer.append(tagName);
+    localDataFileWriteBuffer.append(LocalDataFileConstants.LOCAL_DATA_FILE_DELIMITER);
+
+    // Append tag type and delimiter
+    localDataFileWriteBuffer.append(getTagTypeAsString(tagType));
+    localDataFileWriteBuffer.append(LocalDataFileConstants.LOCAL_DATA_FILE_DELIMITER);
+
+    // Append tag value and delimiter
+    if (tagType == TagType.STRING) {
+      localDataFileWriteBuffer.append("\"").append(tagValue).append("\"");
+    } else {
+      localDataFileWriteBuffer.append(tagValue);
+    }
+    localDataFileWriteBuffer.append(LocalDataFileConstants.LOCAL_DATA_FILE_DELIMITER);
+
+    // Append data quality and delimiter (not applicable, use filler)
+    localDataFileWriteBuffer.append(LocalDataFileConstants.LOCAL_DATA_FILE_FILLER_VALUE);
+    localDataFileWriteBuffer.append(LocalDataFileConstants.LOCAL_DATA_FILE_DELIMITER);
+
+    // Append alarm type and delimiter
+    localDataFileWriteBuffer.append(alarmType);
+    localDataFileWriteBuffer.append(LocalDataFileConstants.LOCAL_DATA_FILE_DELIMITER);
+
+    // Append alarm status
+    localDataFileWriteBuffer.append(alarmStatus);
+    localDataFileWriteBuffer.append(LocalDataFileConstants.LOCAL_DATA_FILE_LINE_SEPARATOR);
+  }
+
+  /**
+   * Gets a string representation of the specified tag type.
+   *
+   * @param tagType tag type object
+   * @return tag type string
+   */
+  public static String getTagTypeAsString(TagType tagType) {
+    String tagTypeString = LocalDataFileConstants.LOCAL_DATA_FILE_DATA_TYPE_UNKNOWN;
+    if (tagType == TagType.BOOLEAN) {
+      tagTypeString = LocalDataFileConstants.LOCAL_DATA_FILE_DATA_TYPE_BOOLEAN;
+    } else if (tagType == TagType.DWORD) {
+      tagTypeString = LocalDataFileConstants.LOCAL_DATA_FILE_DATA_TYPE_DWORD;
+    } else if (tagType == TagType.FLOAT) {
+      tagTypeString = LocalDataFileConstants.LOCAL_DATA_FILE_DATA_TYPE_FLOAT;
+    } else if (tagType == TagType.INTEGER) {
+      tagTypeString = LocalDataFileConstants.LOCAL_DATA_FILE_DATA_TYPE_INTEGER;
+    } else if (tagType == TagType.STRING) {
+      tagTypeString = LocalDataFileConstants.LOCAL_DATA_FILE_DATA_TYPE_STRING;
+    }
```

Comment:
can you use a switch here?

Reply:
Incompatible types. Found: 'com.hms_networks.americas.sc.datapoint.DataType', required: 'byte, char, short or int'

(or Enum in Java 1.5+ as well)

Reply:
Since these are really just bytes under the hood cant you just cast?

Reply:
are at least what defines them 
```
  /** Byte assigned to represent the boolean data type. */
  private static final byte DATA_TYPE_BOOLEAN = 0;

  /** Byte assigned to represent the float data type. */
  private static final byte DATA_TYPE_FLOAT = 1;

  /** Byte assigned to represent the integer data type. */
  private static final byte DATA_TYPE_INTEGER = 2;

  /** Byte assigned to represent the DWORD type. */
  private static final byte DATA_TYPE_DWORD = 3;

  /** Byte assigned to represent the string data type. */
  private static final byte DATA_TYPE_STRING = 4;
```

Reply:
TagType.BOOLEAN, TagType.DWORD, etc are instances of the TagType object under the hood. It can't be compared as a byte.

Reply:
> TagType.BOOLEAN, TagType.DWORD, etc are instances of the TagType object under the hood. It can't be compared as a byte.

```
getRawDataType()
```

Reply:
The byte is only what is used under the hood to instantiate (and differentiate) the object. Since these 'enum like classes' are basically enums, they cant be compared with anything but itself

Reply:
This would require making the DATA_TYPE_xxxxx constants public though.

Reply:
Yeah its probably fine as is, just an annoying thing that makes the code much worse to look at.

Reply:
I cannot recall why getRawDataType() is even there (no calls in Azure Connector). I think using that defeats the purpose of having the enum-like classes. Additionally, the bytes are still private in that class, so another copy of those bytes as constants would need to be added.

Reply:
I do agree, it's not ideal as far as formatting/style. I really wish we just had native enums :(